### ISSUE TITLE
What’s the derivation of this code?

### ISSUE THREAD

**Issue by CPMArchives** [sergev/colossal-cave-adventure-ru#1]
I’m an historian of Crowther and Woods’ Colossal Cave. I maintain a family tree of variants at [http](https://mipmip.org/advfamily/advfamily.html). I’m trying to determine the ancestry of this version of Adventure. The Readme mentions code from Taplap.ru, and DemOS and MS-DOS versions, and the C code version here is for the UNIX platform. However, this leaves some questions. Based on the treasures and items lists, this is not just a translation of the original Crowther and Woods game, but extends it. Do you know the source of the Tsplsp.ru code, or the connection to the MS-DOS version mentioned in the notes? Thanks.

-nathanael

**Comment by leobru** [comment]
Versions containing the mistranslation "Хороший дом для большого родника" ("A good house for a large spring", instead of "well-house") most likely stem from the early 1980s (no later than 1983) translation of the 350-point Fortran version, done by Vadim V. Kobelev of IPMCE (reportedly in 3 or 5 days on a bet) for the BESM-6, which contained the error. That translation was quite faithful; later variants for the Eastern-bloc DEC clones may contain "creative" modifications.

**Comment by sergev** [comment]
These sources seem to be an enhanced version of ANON0340. Compared to ftp://ftp.simtreas.ru/pub/download/add/adventur.tgz, there are only minor changes in `base/advcase` file.

**Comment by CPMArchives** [comment]
Thank you both for your responses, and sorry for the continued questions.  I believe the following to be true:

Kobelev adapted Wood to BESM-6/Fortran.
Someone (but who?) adapted Kobelev to the SM-4/DemOS/Fortran.
Lukhno ported the DemOS version to C for MS-DOS, adding expansions and enhancements.
Vakulenko adapted Lukhno for UNIX.
The code posted to Taplap.ru is just Lukhno.
ANON0340 is just a snapshot of Vakulenko.

Are there any corrections to the above? Are Kobelev and the DemOS version now lost?

**Comment by sergev** [comment]
Not exactly. Looks like there were two independent translations (and ports) of Colossal Cave (by Wood) to Russian.

(1) Kobelev ported Wood to BESM-6 (in Russian), approximately in late 70s. This version is lost. I played it a lot in late 80s. I have a map of it.

(2) I investigated ANON0340 sources and found that this is another port, by [Vadim Antonov](https://en.wikipedia.org/wiki/Vadim_Antonov), probably 1984. In file `adv_cave` you can find:
```
l                WELCOME TO THE HACKERDOME !
l                      ( @VG private )
```
This **`@VG`** is a unique signature of Vadim Antonov. It means Antonov Vadim Gennadievich, with A replaced by @.

Sources of ANON0340 came from FTP site simtreas.ru. On the same site I found sources of another game ported by Antonov: Star Trek. Here is the link: ftp://ftp.simtreas.ru/pub/download/add/TREK.tgz

When unpacked and converted from koi8-r to utf-8, in file `readme` we can see:
```
Запущено на СМ-4 в МНОС РЛ 1.2    29.12.84
                            Антонов В.Г. ( ВМК МГУ, ИПК Минавтопрома )
Им же переведено на русский язык.
```
In English:
```
Ported to SM-4 in MNOS RL 1.2 12/29/84 
by Antonov V.G. (VMK MSU, IPK Ministry of Automotive Industry) 
Also translated into Russian by him.
```
So this StarTrek came from Antonov. I'm pretty sure the Adventure is also his translation. Note that Antonov not only ported the game, but also substantially modified the map, especially at the game end.

Version from Lukhno is mostly the same as Antonov, with minor changes in the game database, and better C code. I enhanced it a little bit, mostly for UTF-8 encoding.

**Comment by NCJECulver** [comment]
Thanks for the reply. I was finally able to access simtreas.ru and download adventur.tgz.

I was under the impression the BESM-6 and the SM-4 didn't support C. Yet Antonov is C.

So:
Kobelev adapted Wood to BESM-6. Antonov adapted Wood to SM-4/DemOS.
Lukhno adapted Antonov to MS-DOS.
Vakulenko is adapting Lukhno to UNIX.
The extensions were added by Antonov and retained by Lukhno.
ANON0340 is Antonov.

As to dates, Kobelev was late '70s, Antonov was 1985 and Lukhno was 1997(?). It seems likely, then, that the version Lukhno reports seeing on the SM-4/DemOS was Antonov.

I'm curious what to make of leobru's comment above about the "good house" mistranslation originating in Kobelev. I find it in both Antonov (ANON0340) and Lukhno.

**Comment by CPMArchives** [comment]
Thanks for the reply. I was finally able to access simtreas.ru and download adventur.tgz.

I was under the impression the BESM-6 and the SM-4 didn't support C. Yet Antonov is C. Or is Antonov not written for the SM-4, and the version Lakhno reports seeing in 1986 is yet another version? [UPDATE: ah, never mind. I missed your note: "Ported to SM-4 in MNOS RL". So perhaps we can surmise Lakhno is misremembering the DemOS bit.]

So:
Kobelev adapted Wood to BESM-6. Antonov adapted Wood to SM-4/DemOS.
Lakhno adapted Antonov to MS-DOS.
Vakulenko is adapting Lukhno to UNIX.
The extensions were added by Antonov and retained by Lakhno.
ANON0340 is Antonov.

As to dates, Kobelev was late '70s, Antonov was 1985 and Lakhno was 1997(?). It seems likely, then, that the version Lakhno reports seeing on the SM-4/DemOS was Antonov.

Do you remember how many points Kobelev had? 350?

I'm curious what to make of leobru's comment above about the "good house" mistranslation originating in Kobelev. I find it in both Antonov (ANON0340) and Lakhno.

**Comment by leobru** [comment]
Kobelev had 350 points.

Perhaps people were taking the translated text database from the BESM-6
variant, even when they had to adapt the program part independently.

Leo

**Comment by leobru** [comment]
Speaking of the Kobelev's BESM-6 version, there is a chance that it is
present on one of the disks at the London Science Museum, which they have
nebulous plans to read but can never find time or resources for it.

Leo

**Comment by sergev** [comment]
BTW, there is another Russian translation of Colossal Cave to Russian: by Denis Gaev in 2003.
Here are the sources: https://github.com/sergev/adventure-ru-inform/blob/main/AdventR.inf

**Comment by leobru** [comment]
Now that's an independent translation. It rephrases "a well-house for a
large spring" as "the source of the spring is inside the house".

Leo

**Comment by CPMArchives** [comment]
This appears to be a Russian translation of Graham Nelson's NELS0350.
